File: 백준/Silver/1064. 평행사변형/평행사변형.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int x1 = sc.nextInt();
        int y1 = sc.nextInt();
        int x2 = sc.nextInt();
        int y2 = sc.nextInt();
        int x3 = sc.nextInt();
        int y3 = sc.nextInt();

        double d1 = distance(x2, y2, x3, y3);
        double d2 = distance(x3, y3, x1, y1);
        double d3 = distance(x1, y1, x2, y2);

        // 세 점이 일직선인 경우, 평행사변형을 만들 수 없는 경우 처리
        if ((x2 - x1) * (y3 - y1) == (y2 - y1) * (x3 - x1)) {
            System.out.println("-1");
        } else {
            // 최댓값과 최솟값의 차이를 출력
            double result = (Math.max(d1, Math.max(d2, d3)) - Math.min(d1, Math.min(d2, d3))) * 2;
            System.out.printf("%.16f%n", result);
        }

        sc.close();
    }

    // 두 점 사이 거리 계산
    private static double distance(int x1, int y1, int x2, int y2) {
        return Math.hypot(x1 - x2, y1 - y2);
    }
}

File: 백준/Silver/1064. 평행사변형/평행사변형.cc
#include <iostream>
#include <cmath>
#include <iomanip>
#include <algorithm>

using namespace std;

struct Point {
    double x, y;
};

double getLength(Point a, Point b) {
    return hypot(a.x - b.x, a.y - b.y);
}

// 두 벡터 (a -> b)와 (b -> c)가 평행한지 여부 확인
bool isCollinear(Point a, Point b, Point c) {
    return (b.x - a.x) * (c.y - a.y) == (b.y - a.y) * (c.x - a.x);
}

int main() {
    Point a, b, c;
    cin >> a.x >> a.y >> b.x >> b.y >> c.x >> c.y;

    // 세 점이 일직선 상에 있는지 확인
    if (isCollinear(a, b, c)) {
        cout << "-1.0" << endl;
        return 0;
    }

    // 각각의 변 길이 계산
    double abLen = getLength(a, b);
    double bcLen = getLength(b, c);
    double acLen = getLength(a, c);

    // 세 개의 평행사변형 둘레 계산
    double len1 = 2 * (abLen + bcLen);
    double len2 = 2 * (bcLen + acLen);
    double len3 = 2 * (abLen + acLen);

    // 최대 둘레와 최소 둘레의 차이 계산 및 출력
    double result = max({len1, len2, len3}) - min({len1, len2, len3});
    cout << fixed << setprecision(16) << result << endl;

    return 0;
}
